apps/dh/api-dh/source/DataHub.WebApi/Modules/ElectricityMarket/MeteringPointNode.cs

```diff
@@ -64,13 +77,38 @@ public static async Task<RelatedMeteringPointsDto> GetRelatedMeteringPointsAsync
             string meteringPointId,
             CancellationToken ct,
             [Service] IElectricityMarketClient_V1 client) =>
-                await client.MeteringPointRelatedAsync(meteringPointId, ct).ConfigureAwait(false);
+                await client.MeteringPointRelatedAsync(meteringPointId, null, ct).ConfigureAwait(false);
 
     [Query]
     [Authorize(Roles = new[] { "metering-point:search" })]
     public static async Task<MeteringPointDto> GetMeteringPointAsync(
         string meteringPointId,
         CancellationToken ct,
-        [Service] IElectricityMarketClient_V1 client) =>
-            await client.MeteringPointAsync(meteringPointId, ct).ConfigureAwait(false);
+        [Service] IHttpContextAccessor httpContextAccessor,
+        [Service] IRequestAuthorization requestAuthorization,
+        [Service] AuthorizedHttpClientFactory authorizedHttpClientFactory)
+    {
+        if (httpContextAccessor.HttpContext == null)
+        {
+            throw new InvalidOperationException("Http context is not available.");
+        }
+
+        var user = httpContextAccessor.HttpContext.User;
+
+        var actorNumber = user.GetActorNumber();
+        var marketRole = Enum.Parse<EicFunctionAuth>(user.GetActorMarketRole());
```

Comment:
Suggestion: Make a extension method like this one apps/dh/api-dh/source/DataHub.WebApi/Modules/ElectricityMarket/Extensions/MeteringPointDtoExtensions.cs to map EicFunction from one subsystem to another. So we dont need up with an exception because they are out of sync